### DISCUSSION TITLE
accessing multiple concourse setups behind haproxy

### DISCUSSION THREAD

**Discussion by HemangLavana** [concourse/concourse#6966] [Help & Support]
Hi,

We have multiple concourse setups running behind haproxy and each instance is accessed via same host but different port. For ex: 

- https://example.com:10001
- https://example.com:10002
- https://example.com:10003

When I try to access these instances from my browser, it will only allow me to access one of them at a time. I cannot have each instance open in a different tab. The moment I log into one instance, it logs me out of the other two instances. So as I switch between the various instances, I have to keep logging back in.

The only other way I found to keep the sessions active is to use different browsers to access each instance (chrome, firefox, safari, etc). Is there a way to configure each concourse instance such that it doesn't share the sessions? I tried passing a different value for --client-id parameter but that didn't help.

Another question: I want to avoid use of port numbers for each instance. For ex:

- https://example.com/project1
- https://example.com/project2
- https://example.com/project3

 I can configure haproxy to select correct concourse instance based on path specified. But concourse doesn't seem to work well a sub-path. I tried specifying the external url for each concourse as per above but that doesn't seem to work.

Thanks,
Hemang.

**Comment by taylorsilva** [comment]
Not too sure about this, but I'm guessing cookies are being over-written since all of your instances share the same hostname and cookies are tied to a hostname, with no distinction given for the port.

Having sub-paths is the most reasonable path forward imo. Sounds like work would need to be done to have this feature supported since your testing indicates Concourse doesn't understand sub-paths after the hostname. Makes sense for it to be built on top of the existing external url flag 👍

**Reply by taylorsilva** [reply]
Looking at a few places in the code, it _appears_ that sub-paths should work. Where exactly were sub-paths not working well in your testing?

**Reply by HemangLavana** [reply]
When I use sub-paths, the home page comes up blank. ok, found a related issue: https://github.com/concourse/concourse/issues/4115

Moreover, having sub-path is not going to solve the cookies' sharing issue if it only depends on the hostname. It would be great if the cookies are based on external url value (so that it would work for any of these: host, host:port, host/sub-path, or host:port/sub-path)

**Reply by taylorsilva** [reply]
looks like #942 logged most work done on supporting this feature. It appears it's mostly the web UI that needs to be updated. Guess that work was never finished.

**Reply by aoldershaw** [reply]
I did some preliminary work to make it easier to implement this in #5301 and #5324 by centralizing the construction of URLs for API calls and assets (meaning there's only a couple places that need to be modified to prefix all requests - at least on the Elm side of things).

However, there's still some other places that URLs are constructed outside of the Elm - not remembering all of them off-hand
